components/proxy/src/main/kotlin/com/hotels/styx/services/YamlFileConfigurationService.kt

```diff
@@ -91,6 +99,7 @@ internal class YamlFileConfigurationService(
         }.mapCatching { (healthMonitors, routingObjectDefs) ->
             updateRoutingObjects(routingObjectDefs)
             updateHealthCheckServices(serviceDb, healthMonitors)
+            originsConfig = content
```

Comment:
It would be better to set the variables in `onSuccess` block like so:

```
        }.onSuccess {
            originsConfig = content
            initialised.countDown()
        }.onFailure {
            LOGGER.error("Failed to reload new configuration. cause='{}'", it.message, it)
        }
```

Reply:
Ah, makes sense. Thanks.